### DISCUSSION TITLE
Idea: Generic modules (i.e nvidia and such) for non-ublue based images

### DISCUSSION THREAD

**Discussion by oomfiee** [blue-build/.github#27] [General]
The idea is to make Blue-build more usable for non-ublue based images (i.e building directly from fedora OCI images).

What i basically think of is blue-build having a module to basically feature those cool stuff from ublue but not needing to base on ublue. Like providing new drivers that arent in fedora or ublue.

The modules will be just simple and the user will have to put what they want in it And fetch the sources if possible. If its not included in blue-build.

Example:

```yml
type: hardware
vendor: nvidia
```

**Comment by xynydev** [comment]
BlueBuild aims to maintain compatibility with non-ublue images. Most modules are even built in such a way that in the future when we would support basing on VanillaOS etc., they would just work.

You speak in a pretty generic way, and I agree on the most part.

The Nvidia example could already be done like this:
```yaml
type: akmods
nvidia: true
install: 
- nvidia 
```
This will use a [ublue-signed-and-built akmod](https://github.com/ublue-os/akmods) and work with any Fedora Atomic base with the default kernel. Since Universal Blue already provides the possibility to use their akmods without using their base images, I don't see any reason to lessen the dependency on them in this aspect. Building and signing Nvidia driver akmods isn't necessarily quick and simple and it's not something I would like to start doing myself, because right now we're getting an equivalent result _for free_. Not to even go into the entirely different problem of Nvidia drivers on other distributions.

If you have any other examples you'd like to be supported, you can share them with us. 

Besides, I don't see any huge reason to use vanilla Fedora images instead of Universal Blue, and if ublue provides some hardware enablement you need, why not just use that?
